head/firmware/main_rev1.cpp

```diff
@@ -0,0 +1,298 @@
+    motor_queue_right("Motor Queue Right");
+
+/**
+ * @brief SPI MSP Initialization
+ * This function configures SPI for the Z/A axis motors
+ * @param hspi: SPI handle pointer
+ * @retval None
+ */
+
+spi::hardware::SPI_interface SPI_intf2 = {
+    .SPI_handle = &hspi2,
+};
+static spi::hardware::Spi spi_comms2(SPI_intf2);
+
+spi::hardware::SPI_interface SPI_intf3 = {
+    .SPI_handle = &hspi3,
+};
+static spi::hardware::Spi spi_comms3(SPI_intf3);
+
+struct motor_hardware::HardwareConfig pin_configurations_left {
+    .direction =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_1,
+            .active_setting = GPIO_PIN_RESET},
+    .step =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_0,
+            .active_setting = GPIO_PIN_SET},
+    .enable =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_4,
+            .active_setting = GPIO_PIN_SET},
+    .limit_switch =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOB,
+            .pin = GPIO_PIN_7,
+            .active_setting = GPIO_PIN_SET},
+    .led = {},
+    .sync_in = {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+        .port = GPIOA,
+        .pin = GPIO_PIN_8,
+        .active_setting = GPIO_PIN_RESET}
+};
+
+struct motor_hardware::HardwareConfig pin_configurations_right {
+    .direction =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_7,
+            .active_setting = GPIO_PIN_RESET},
+    .step =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOC,
+            .pin = GPIO_PIN_6,
+            .active_setting = GPIO_PIN_SET},
+    .enable =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOB,
+            .pin = GPIO_PIN_11,
+            .active_setting = GPIO_PIN_SET},
+    .limit_switch =
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+            .port = GPIOB,
+            .pin = GPIO_PIN_9,
+            .active_setting = GPIO_PIN_SET},
+    .led = {},
+    .sync_in = {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+        .port = GPIOA,
+        .pin = GPIO_PIN_8,
+        .active_setting = GPIO_PIN_RESET}
+};
+
+// TODO clean up the head main file by using interfaces.
+static tmc2160::configs::TMC2160DriverConfig motor_driver_configs_right{
+    .registers =
+        {
+            .gconfig = {.en_pwm_mode = 1},
+            .ihold_irun = {.hold_current = 0x3,
+                           .run_current = 0xe,
```

Comment:
We should set these to 31 & 16 respectively.

Reply:
wait, other way around, right? hold at 16, run at 31